x-pack/plugins/enterprise_search/server/lib/enterprise_search_http_agent.test.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { mockReadFileSync } from '../__mocks__/fs.mock';
+
+import http from 'http';
+import https from 'https';
+
+import { ConfigType } from '../';
+
+import { entSearchHttpAgent, loadCertificateAuthorities } from './enterprise_search_http_agent';
+
+describe('entSearchHttpAgent', () => {
+  it('should be an http.Agent when host URL is using HTTP', () => {
+    const httpAgent = entSearchHttpAgent({ host: 'http://example.org' } as ConfigType);
+    expect(httpAgent instanceof http.Agent).toBe(true);
+  });
+
+  it('should be an http.Agent when host URL is invalid', () => {
+    const httpAgent = entSearchHttpAgent({ host: '##!notarealurl#$', ssl: {} } as ConfigType);
+    expect(httpAgent instanceof http.Agent).toBe(true);
+  });
+
+  it('should be an https.Agent when host URL is using HTTPS', () => {
+    const httpAgent = entSearchHttpAgent({ host: 'https://example.org', ssl: {} } as ConfigType);
```

Comment:
What happens when the host is `https://example.org` but doesn't have `config.ssl` set? I assume that new config is optional, correct?

Reply:
I should rephrase this question to be a little clearer, sorry. Right now what I'm sauing is that if this is the case:

```js
    const httpAgent = entSearchHttpAgent({ host: 'https://example.org' } as ConfigType);
```

the helper returns a non-https agent:

```js
    expect(httpAgent instanceof https.Agent).toBe(false);
```

Is that correct or expected? Or should we expect all `https://` protocol hosts to use `https.Agent` regardless of whether config `ssl` is set?

Reply:
I've made a change to always assume config.ssl will be a thing since it is not optional and some kind of object will always exist there.

Reply:
Thanks a lot, that commit helped clarify things for me!